File: frontend/src/lib/onboarding.ts
// Shared onboarding payload/result types (mirrors backend OnboardingSubmit).

export interface OnboardingAgreement {
  age_confirmed: boolean
  terms: boolean
  privacy: boolean
}

export interface OnboardingConsents {
  data_processing: boolean
  ai_memory: boolean
  notifications_whatsapp: boolean
  notifications_email: boolean
}

export interface OnboardingPayload {
  name: string
  country: string
  age: number
  gender: string
  gender_self_describe?: string | null
  relationship_status: string
  tried_therapy: boolean
  financial_situation: string
  referral_source: string
  referral_other?: string | null
  occupation?: string | null
  concerns: string[]
  concerns_other?: string | null
  support_system: string
  medication: string
  preferred_language: string
  preferred_languages: string[]
  preferred_language_other?: string | null
  therapist_gender_preference: string
  past_therapy_note?: string | null
  therapist_should_know?: string | null
  whatsapp_number?: string | null
  agreement: OnboardingAgreement
  consents: OnboardingConsents
  suitability_none_apply: boolean
}

export interface OnboardingResult {
  onboarding_completed: boolean
  suitability_flagged: boolean
}

File: backend/app/schemas/onboarding.py
"""Schemas for seeker onboarding submission (POST /onboarding)."""

from __future__ import annotations

from pydantic import BaseModel, Field


class Agreement(BaseModel):
    age_confirmed: bool
    terms: bool
    privacy: bool


class ConsentChoices(BaseModel):
    data_processing: bool
    ai_memory: bool = False
    notifications_whatsapp: bool = False
    notifications_email: bool = False


class OnboardingSubmit(BaseModel):
    # Identity / structured answers
    name: str = Field(min_length=1, max_length=80)
    country: str = Field(min_length=2, max_length=2)
    age: int = Field(ge=0, le=120)
    gender: str = Field(max_length=40)
    gender_self_describe: str | None = Field(default=None, max_length=80)
    relationship_status: str = Field(max_length=40)
    tried_therapy: bool
    financial_situation: str = Field(max_length=40)
    referral_source: str = Field(max_length=40)
    referral_other: str | None = Field(default=None, max_length=120)
    occupation: str | None = Field(default=None, max_length=120)
    concerns: list[str] = Field(default_factory=list, max_length=20)
    concerns_other: str | None = Field(default=None, max_length=200)
    support_system: str = Field(max_length=40)
    medication: str = Field(max_length=40)
    preferred_language: str = Field(max_length=40)
    preferred_languages: list[str] = Field(default_factory=list, max_length=10)
    preferred_language_other: str | None = Field(default=None, max_length=80)
    therapist_gender_preference: str = Field(max_length=40)

    # Sensitive free-text (optional) — never logged.
    past_therapy_note: str | None = Field(default=None, max_length=2000)
    therapist_should_know: str | None = Field(default=None, max_length=2000)
    whatsapp_number: str | None = Field(default=None, max_length=20)

    agreement: Agreement
    consents: ConsentChoices

    # true = none of the out-of-scope conditions apply; false = one or more apply.
    suitability_none_apply: bool


class OnboardingResult(BaseModel):
    onboarding_completed: bool
    suitability_flagged: bool
